cfep-9-migrator-file.md

```diff
@@ -0,0 +1,209 @@
+# @mariusvniekerk's proposal
+
+This is a general sketch of what I'd like migrators to be able to achieve.  
+Effectively this means we can decompose all of our migrations into a small set of primitives 
+that we can apply to generate a resulting desired target.
+
+----
+
+Given a feedstock with the following pinnings
+
+```yaml
+# Existing pins
+a:
+  - 1.5
+b:
+  - 1.2
+someflag:
+  - disabled
+```
+
+There are four classes of pinning we can do in general.
+
+## 1. Simple Additive Version Migrations
+
+<table>
+<tr>
+<td>
+
+```yaml
+# Existing pins
+a:
+  - 1.5
+b:
+  - 1.2
+someflag:
+  - disabled
+```
+</td>
+<td> with </td>
+<td>
+
+```yaml
+# migrator
+__migrator: kind: version
+a:
+  - 1.6
+b:
+  - 1.1
+```
+</td>
+<td> -> </td>
+<td>
+
+```yaml
+# resulting pins
+a:
+  - 1.6
+b:
+  - 1.2
```

Comment:
Why do versions always increase? This assumption is limiting in a way that is not required.

Reply:
Mostly because for migrations decreases are almost certainly an antipattern